lib/tree.rb

```diff
@@ -1,38 +1,77 @@
 class NoApplesError < StandardError; end
 
+
+
 class AppleTree
-  attr_#fill_in :height, :age, :apples, :alive
+  attr_reader :height, :age, :apples, :alive
 
-  def initialize
+  def initialize(height, age, apples, alive)
+    @height = height
+    @age = age
+    @apples = apples
+    @alive = alive
   end
 
   def age!
+    @age = @age + 1
+    @height = @height + Random.new.rand(0.5..2.5)
+    self.add_apples
   end
 
   def add_apples
+    rn = Random.new
+    numApples = rn.rand(1..10)
+    diameter = rn.rand(2.0..4.0)
+
+    for i in 0..numApples
+      apples.push(Apple.new("Red", diameter))
+    end
   end
 
   def any_apples?
+    if(@apples == nil || @apples.count <= 0)
```

Comment:
why might we want to use the getter here instead and anywhere else we can in this class?